### DISCUSSION TITLE
Anonymous Login

### DISCUSSION THREAD

**Discussion by FShinoda** [FreeOpcUa/opcua-asyncio#1275] [Q&A] [answered]
Hello there! 

I've been trying to login to my client's opc-ua server via certificate and user credentials but they say I am autheticating as anonymous. 
Am I doing something wrong? Or maybe it is a bug?

My code:
![image](https://user-images.githubusercontent.com/45886482/230904502-6bcddbf8-4dd4-4843-ad3e-af0ae4d7811f.png)

The server log: it returns success login but as anonymous
![image](https://user-images.githubusercontent.com/45886482/230904587-b57e5af2-725a-483d-a8ae-cabae2d6f19b.png)

**Comment by schroeder-** [comment]
Why you call open_securechannel ? This call can mess with everything.

**Reply by FShinoda** [reply, marked as answer]
Oddly, I've just fixed that by running my code without user and password (logging permission denied), and then, adding the credentials back again. Thank you for your fast answer!

Additionally, can you describe more about how the `open_secure_channel` call can be problematic?

**Reply by schroeder-** [reply]
Because open_secure_channel is already called in the `async with client` and a second call can lead to errors.

**Reply by FShinoda** [reply]
Mmm, got it. So when it would be appropriate to call `open_secure_channel` ?

**Reply by schroeder-** [reply]
Never. The internal code call open_secure_channel. Every opc ua connection opens a secure channel, so no need to call from user code at all.

**Reply by FShinoda** [reply]
ok, thanks a lot!
